packages/x-charts/src/internals/plugins/featurePlugins/useChartCartesianAxis/useChartCartesianAxis.types.ts

```diff
@@ -84,6 +98,10 @@ export interface UseChartCartesianAxisState {
     x: DefaultedXAxis[];
     y: DefaultedYAxis[];
   };
+  /**
+   * The controlled axis item highlighted.
+   */
+  controlledCartesianAxisHighlight?: AxisItemIdentifier | null;
```

Comment:
Can't we have more than one highlighted axis item?

Reply:
In theory it's not that hard. In practice I don't see usecases.

I can think about the crosshair but when following the point, I feel doing store update would be a bad trade off in terms of performances. Bette to do all the computation in the crosshair component itself.

![image](https://github.com/user-attachments/assets/edbd0c41-4035-4c67-ac0b-2c9554da4217)


I can also think about someone who would like to highlight multiple values at once. But in that case, it would probably be better to create his own highlight component

Reply:
> In theory it's not that hard. In practice I don't see usecases.

What about charts with multiple X axes? Can't there be more than one axis highlight at a time?

> I can also think about someone who would like to highlight multiple values at once. But in that case, it would probably be better to create his own highlight component

Would that be something that's easy to do? Should we have a demo for it?

I suspect it's somewhat useful. For example, you might want to click a button and highlight some points.

Reply:
> For me it looks like the worst case scenario with two axis highligh component rendered either on top of each other or fairly close

Sure, but shouldn't we support it? I think it's a bit weird that in controlled state the handler provides an array of values and the state itself only accepts a single value. IMO, both should accept an array or a single value. 

In this case, since there can be multiple axes, I think we should always use an array. What downsides do you see in this?